futures/src/client.rs

```diff
@@ -202,12 +197,17 @@ impl<T: AsyncRead+AsyncWrite+Send+'static> Client<T> {
   /// );
   /// # }
   /// ```
-  pub fn connect(stream: T, options: ConnectionOptions) -> impl Future<Item = (Self, Heartbeat), Error = io::Error> + Send + 'static {
+  pub fn connect(stream: T, options: ConnectionOptions) ->
+    impl Future<Item = (Self, Heartbeat<impl Future<Item = (), Error = io::Error>>), Error = io::Error> + Send + 'static
```

Comment:
Maybe add "+ Send + 'static" in the Heartbeat type param, just for consistency's sake?